ibm/resource_ibm_satellite_cluster.go

```diff
@@ -0,0 +1,602 @@
+// Copyright IBM Corp. 2017, 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package ibm
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	v1 "github.com/IBM-Cloud/bluemix-go/api/container/containerv1"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.ibm.com/ibmcloud/kubernetesservice-go-sdk/kubernetesserviceapiv1"
+)
+
+const (
+	isLocationProvisioning = "provisioning"
+	isLocationNormal       = "normal"
+	isCluterDeploying      = "deploying"
+	isCluterDeployFailed   = "deploy_failed"
+	isClusterNormal        = "warning"
+	isClusterDeleting      = "deleting"
+	isClusterDeleteDone    = "done"
+	isWorkderDeployed      = "deployed"
+)
+
+func resourceIBMSatelliteCluster() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceIBMSatelliteClusterCreate,
+		Read:     resourceIBMSatelliteClusterRead,
+		Update:   resourceIBMSatelliteClusterUpdate,
+		Delete:   resourceIBMSatelliteClusterDelete,
+		Importer: &schema.ResourceImporter{},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Read:   schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+			Update: schema.DefaultTimeout(60 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The unique name for the new IBM Cloud Satellite cluster",
+			},
+			"location": {
+				Type:        schema.TypeString,
+				Required:    true,
```

Comment:
I believe location should be a force New resource

Reply:
location attribute is immutable now.